src/string/utf16.cr

```diff
@@ -12,22 +12,40 @@ class String
   # "hi 𐂥".to_utf16 # => Slice[104_u16, 105_u16, 32_u16, 55296_u16, 56485_u16]
   # ```
   def to_utf16 : Slice(UInt16)
-    size = 0
+    if ascii_only?
+      # size == bytesize, so each char fits in one UInt16
+
+      # This is essentially equivalent to `to_slice.map(&.to_u16)` but also makes
+      # sure to allocate a null byte after the string.
+      slice = Slice(UInt16).new(bytesize + 1) do |i|
+        if i == bytesize
+          0_u16
+        else
+          unsafe_byte_at(i).to_u16
+        end
+      end
+      return slice[0, bytesize]
```

Comment:
Isn't this chopping off the null byte? Why can't `slice` be returned?

Reply:
That's the same as the  non-ascii behaviour. The method description explicitly states that the null byte is not part of the returned slice.

Reply:
Oh, interesting. I guess it's fine. A bit strange, but fine :-)